build-scripts/build_compiler.js

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2018 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Build the compiler java jar and GWT platforms from source.
+ *
+ * Invoked as part of the package build process:
+ *
+ *     yarn run build
+ *
+ * For pull requests and pushes to master, the compiler submodule is expected to be pointed at the tagged commit
+ * that matches the package major version.
+ *
+ * When the COMPILER_NIGHTLY env variable is set, the compiler submodule will be pointing to the latest master
+ * commit. This is for regular integration testing of the compiler with the various tests in this repo's packages.
+ * In this case the compiler will be built as a SNAPSHOT.
+ */
+'use strict';
+
+const {spawn, spawnSync} = require('child_process');
+const ncp = require('ncp');
+const Semver = require('semver');
+
+// Master version number of the packages. Replaces the version
+// information from the root package.json file.
+const packageInfo = require('../lerna.json');
+const packageVersion = new Semver(packageInfo.version);
+
+const fs = require('fs');
+const glob = require('glob');
+
+function getCompilerSubmoduleReleaseVersion() {
+  const gitTagResults = spawnSync('git', ['tag', '--points-at', 'HEAD'], {
+    cwd: './compiler'
+  });
+  if (gitTagResults.status === 0) {
+    const currentTag = gitTagResults.stdout.toString().replace(/\s/g, '');
+    let normalizedTag = currentTag;
+    if (normalizedTag) {
+      // Standard release tags are of the form "vYYYYMMDD".
+      // We also recognize the forms where a special release name is present:
+      //     "webpack-v20180810"
+      //     "v20180810-gwt-fix"
+      normalizedTag = currentTag.replace(/^([a-z]+-)?(v\d{8})(.*)$/, '$2');
+    }
+    return normalizedTag;
+  }
```

Comment:
Please add an explicit else/return case.

Reply:
This logic has now been removed.